marathon-bigip-ctlr.py

```diff
@@ -577,10 +577,17 @@ def get_apps(apps, health_check):
                 marathon_app.app['labels']['F5_PARTITION']
         marathon_apps.append(marathon_app)
 
-        service_ports = app['ports']
-        logger.debug("Application service ports = %s", (repr(service_ports)))
-        logger.debug("Labels for app %s: %s", app['id'],
-                     marathon_app.app['labels'])
+        # Address nonexistent 'ports' for application when DC/OS Virtual
+        # Networking is used.
+        try:
+            service_ports = app['ports']
+            logger.debug("Application service ports = %s",
+                         (repr(service_ports)))
+            logger.debug("Labels for app %s: %s", app['id'],
+                         marathon_app.app['labels'])
+        except Exception:
```

Comment:
If we are just trying to catch a KeyError exception here, I would use the specific exception rather than blanketing all Exceptions. Not that anything else in this block *should* throw a different exception (maybe the logger), but we don't want to miss a different exception *if* one gets raised.

Reply:
Yes!